### PR TITLE
Pl 69 to 72 summary key functionalities and PL 160

### PR DESCRIPTION
**Pull request by matteovisotto**
DONE:
- TLDR
- Abstract
- Abstract Summary
- Selection summary

NOTE: This is the refactor for PL-69, PL-70, PL-71, PL-72

### CODE REVIEW

File: src/components/PdfReaderComponent.js [hunk 1]
```diff
@@ -96,6 +99,26 @@ class PdfReaderComponent {
...
+        }
+        if (key === "u") {
+          this.#onNewFile();
+        }
+      }
+    );
+  };
+
+  /**
+   * Cretes event triggered when application view changed from reader view to input view.
```
Comment (tonesija) [inline src/components/PdfReaderComponent.js:117]: I know you did not write this, but this jsdoc needs to be written differently.

File: src/components/SelectionPopUpComponent.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import { getCurrentDOMSelection } from "../services/Utils";
+import {
+  EventHandlerService,
+  PDFLEvents,
+} from "../services/EventHandlerService";
+import { POPUP_DISAPPEAR_TIMEOUT } from "../Constants";
+
+class SelectionPopUpComponent {
```
Comment (tonesija) [inline src/components/SelectionPopUpComponent.js:8]: jsdoc

File: src/components/SelectionPopUpComponent.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    this.#handleSelection({
+      x: event.clientX + window.scrollX,
+      y: event.clientY + window.scrollY,
+    });
+  };
+
+  /**
+   * Selection handling function
+   * Get selected text
+   * If a text exists diplasy the popup to generate summary
```
Comment (tonesija) [inline src/components/SelectionPopUpComponent.js:50]: Why If is capitalized and new line?

File: src/components/SidePageComponent.js [hunk 4]
```diff
@@ -35,6 +36,7 @@ class SidePageComponent {
   constructor() {
     this.knowledgeGraphComponent = new KnowledgeGraphComponent();
     this.summaryKeyComponent = new SummaryKeyComponent();
+    this.selectionPopUpComponent = new SelectionPopUpComponent();
```
Comment (tonesija) [inline src/components/SidePageComponent.js:39]: Why is this also here?

File: src/components/SummaryKeyComponents/SummaryKeyComponent.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+  /**
+   * Creates and initializes new Summary Key Component
+   * @constructor
+   */
+  constructor() {
+    this.tldrItem = new TLDRAccordionItem();
+    this.abstractItem = new AbstractAccordionItem();
+    this.abstractSummaryItem = new AbstractSummaryAccordionItem();
+    this.selectionSummaryItem = new SelectionSummaryAccordionItem();
```
Comment (tonesija) [inline src/components/SummaryKeyComponents/SummaryKeyComponent.js:36]: These 4 could have been the same class instantiated 4 times.

File: src/components/SummaryKeyComponents/AbstractAccordionItem.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+class AbstractAccordionItem {
+  components = {
+    abstractText: document.querySelector("#abstract-text"),
+  };
+
+  /**
+   * Set the text of the accordion item
+   * @param text text to be set
+   */
+  setText = (text) => {
```
Comment (tonesija) [inline src/components/SummaryKeyComponents/AbstractAccordionItem.js:16]: setText, setError, setLoading and clear repeat themselves 4 times through these 4 classes.

**Comment by tonesija** [comment]
I still did not run it locally, I'll continue my review soon.

File: src/components/SummaryKeyComponents/SummaryKeyComponent.js (hunk 5 above)
Comment (matteovisotto) [reply src/components/SummaryKeyComponents/SummaryKeyComponent.js:36]: So, maybe give the dom id in the constructor and create just one class?
And at this point since clear, setText and setError are a little bit different from each other I move the logic in the SummaryKeyComponent?

File: src/components/PdfReaderComponent.js (hunk 1 above)
Comment (matteovisotto) [reply src/components/PdfReaderComponent.js:117]: This shouldn't  be here at all, removed now

File: src/components/SidePageComponent.js (hunk 4 above)
Comment (matteovisotto) [reply src/components/SidePageComponent.js]: removed

File: src/components/SelectionPopUpComponent.js (hunk 3 above)
Comment (matteovisotto) [reply src/components/SelectionPopUpComponent.js]: changed

File: src/components/SummaryKeyComponents/SummaryKeyComponent.js (hunk 5 above)
Comment (tonesija) [reply src/components/SummaryKeyComponents/SummaryKeyComponent.js:36]: setText can just take text and set it, so from the outside you take care of how to prepare it, for setError you can pass the error message throught the constructor

**Comment by paolocorsa** [comment]
I tested it locally:
-the popup for summarize appears when it shouldn't (sometimes when you click on one of the section for the summary the popup appears)
Screenshot_20221229_143346
-Selected text summary is good but when you select a portion of text, open the section to visualize the summary and then select a bigger portion of text to summarize it is cutted (i'm not very sure for this)
Screenshot_20221229_143747
-TLDR, Abstract and Abstract summary are good because for every document they are static

**Comment by matteovisotto** [comment]
Good work @paolocorsa 
Can I have more Information about the first point ?
For the text length i know it and it is a CSS problem.

**Review by tonesija — APPROVED** [review]
If everything works it's a +1 from me. We don't have much time and a lot to do. And we also have the refactor task, so if something is wrong we can fix it there.

**Comment by paolocorsa** [comment]
I tested it and all problems that there was in this PR are resolved, for me +1.

**Comment by gabrielekarra** [comment]
this can be merged for sure @matteovisotto
